pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,294 @@
+        map_values=True
+    )
+
+    def save_config(self, index):
+        """Save the current config to memory.
+
+        :param index:
+            string: index of the location to save the configuration
+        """
+        self.save_settings = index
+
+    def recall_config(self, index):
+        """Recall a config from memory.
+
+        :param index:
+            string: index of the location from which to recall the configuration
+        """
+        self.recall_settings = index
+
+
+class SPDChannel(object):
+    """ The channel class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, instrument, channel: int = 1):
+        self.instrument = instrument
+        self.channel = channel
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    current = Instrument.measurement(
+        "MEAS:CURR? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    power = Instrument.measurement(
+        "MEAS:POWE? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    set_current = Instrument.control(
+        "CH{channel}:CURR?", "CH{channel}:CURR %g",
+        """Control the output current configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 8],
+        dynamic=True
+    )
+
+    set_voltage = Instrument.control(
+        "CH{channel}:VOLT?", "CH{channel}:VOLT %g",
+        """Control the output voltage configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 16],
+        dynamic=True
+    )
+
+    output = Instrument.setting(
+        "OUTP CH{channel},%s",
+        """Configure the power supply output.
+
+        :type: bool
+            ``True``: enables the output
+            ``False``: disables the output
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    enable_timer = Instrument.setting(
+        "TIME CH{channel},%s",
+        """Enable the channel timer.
+
+        :type: bool
+            ``True``: enables the timer
+            ``False``: disables it
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    def ask(self, cmd):
+        return self.instrument.ask(cmd.format(channel=self.channel))
+
+    def write(self, cmd):
+        self.instrument.write(cmd.format(channel=self.channel))
+
+    def values(self, cmd, **kwargs):
+        """ Reads a set of values from the instrument through the adapter,
+        passing on any key-word arguments.
+        """
+        return self.instrument.values(cmd.format(channel=self.channel), **kwargs)
```

Comment:
If you can wait for some time until #718 lands, the whole channel handling will become much easier, with less need to define these methods.

Reply:
I agree, this seems ready pending the channel changes. 
I would say wait until #718 lands, merge a fresh master into your branch, then rework. 
Alternatively, @bmoneke do you want this in master now, pull into #718 and rework this instrument within #718?

Reply:
There are many instruments with channels, which I do not rework in #718.
We can rework them step by step, as they will continue to work as they are (just the code base is not as nice).

Let's merge this work as is, in order to close the PR

Reply:
Sounds good! I opened #746 for tracking this.